### PR TITLE
Hardhat-etherscan verify subtask

### LINKED ISSUE #945: [hardhat-etherscan] Programatically pass complex arguments to verify task
How to verify if my contract constructor has array params?

Let say i have a contract that have constructor accepting 2 params, string and array

````
constructor(string memory param1, address[] memory param2) public {
}
````

I want to verify it using etherscan verify plugin. 

Tried code below it doesn't work.
````
const bre = require("@nomiclabs/buidler");
await bre.run('verify', {
        address: '0xXXXXXX,
        constructorArguments: ['Some params', ['test']],
    });
````
This will result in error `Invalid value  for argument constructorArguments of type string`.

### PR DESCRIPTION
**Pull request by scnale**
This adds a subtask with a friendly API for users that want to verify a contract within a script or task.

Fixes #945.

### CODE REVIEW

File: packages/hardhat-etherscan/src/index.ts [hunk 1]
```diff
@@ -436,9 +472,16 @@ Possible causes are:
 Please use the contract parameter with one of the following contracts:
 ${nameList}
 
-For example:
+For example, if you are using the verify task from CLI:
```
Comment (fvictorio) [inline packages/hardhat-etherscan/src/index.ts:475]:
```suggestion
For example:
```

File: packages/hardhat-etherscan/src/index.ts [hunk 2]
```diff
@@ -436,9 +472,16 @@ Possible causes are:
...
 
-  hardhat verify --contract contracts/Example.sol:ExampleContract <your verify args>`;
+  hardhat verify --contract contracts/Example.sol:ExampleContract <other args>
+
+If you are running the verify subtask from within Hardhat instead:
+
+const { success } = await run("${TASK_VERIFY_VERIFY}", {
```
Comment (fvictorio) [inline packages/hardhat-etherscan/src/index.ts:481]:
```suggestion
await run("${TASK_VERIFY_VERIFY}", {
```

**Review by fvictorio — APPROVED** [review]

**Comment by scnale** [comment]
Fixes #945.

**Comment by fvictorio** [comment]
Closed in favor of #1114

**Pull request closed without merging**